src/main/kotlin/lotto/services/LottoService.kt

```diff
@@ -0,0 +1,51 @@
+package lotto.services
+
+import lotto.domain.Lotto
+import lotto.domain.Rank
+import lotto.domain.WinningCombination
+import lotto.exceptions.LottoException
+
+object LottoService {
+    private const val ZERO_DOUBLE = 0.0
+
+    fun purchaseAmountValidator(amount: Int) {
+        if (amount < Lotto.Companion.PRICE_OF_TICKET || amount % Lotto.Companion.PRICE_OF_TICKET != 0) {
+            throw LottoException.InvalidAmountException(
+                amount,
+            )
+        }
+    }
+
+    fun generateLottoTickets(amount: Int): List<Lotto> {
+        val numberTickets = amount / Lotto.Companion.PRICE_OF_TICKET
+        return List(numberTickets) { getLottoTicket() }
+    }
+
+    fun getRankFrom(
+        ticket: Lotto,
+        winningNumbers: WinningCombination,
+    ): Rank {
+        val countOfMatch = ticket.numbers.count { it in winningNumbers.winningTicket.numbers }
+        val matchBonus = ticket.numbers.any { it == winningNumbers.bonusNumber }
+        return Rank.valueOf(countOfMatch, matchBonus)
+    }
+
+    fun calculateReturnRate(
+        totalAmount: Int,
+        results: List<Rank>,
+    ): Double {
+        val totalPrize = results.sumOf { it.winningMoney }
+        return if (totalPrize == 0) ZERO_DOUBLE else (totalPrize.toDouble() / totalAmount) * 100.0
+    }
+
+    private fun getLottoTicket(): Lotto {
+        val numbers = generateRandomNumbers(Lotto.Companion.LOTTO_PICK_SIZE)
+        return Lotto(numbers)
+    }
+
+    private fun generateRandomNumbers(size: Int): List<Int> {
+        return (Lotto.MIN_RANGE..Lotto.MAX_RANGE)
+            .shuffled()
+            .take(size)
+    }
```

Comment:
```suggestion
    private fun getLottoTicket(): Lotto {
        val numbers = generateRandomNumbers()
        return Lotto(numbers)
    }

    private fun generateRandomNumbers(): List<Int> {
        return (Lotto.MIN_RANGE..Lotto.MAX_RANGE)
            .shuffled()
            .take(Lotto.LOTTO_PICK_SIZE)
    }
```

If the method is used only once and the parameter can be removed, I think it's good to simplify the logic.
When we need to use it somewhere else, that's when we can start refactoring.
I'm curious about your thoughts on this :)

Also, `Companion` can be removed when using constants so please check this in other places as well.

Reply:
You're absolutely right, simplifying generateRandomNumbers() by removing the unused parameter makes the code cleaner, especially since it's only used in one place. I don't know what I was thinking😔.

Good point about the Companion😩. I’ll pay more attention to those details, especially when I’m not the one typing during pair programming. It’s easy to overlook small things like that in the moment. Thanks for pointing it out!